src/Stratis.Bitcoin/Consensus/ConsensusRuleEngine.cs

```diff
@@ -234,6 +233,10 @@ private async Task ExecuteRulesAsync(IEnumerable<AsyncConsensusRule> asyncRules,
             {
                 ruleContext.ValidationContext.Error = ex.ConsensusError;
             }
+            catch (Exception ex)
```

Comment:
Probably not the correct solution, but also I don't think this is the problem to the bug you got.
I would say we need to catch such exception at the validators and handles there